scene/resources/visual_shader_nodes.cpp

```diff
@@ -3085,3 +3085,123 @@ VisualShaderNodeSwitch::VisualShaderNodeSwitch() {
 	set_input_port_default_value(1, Vector3(0.0, 0.0, 0.0));
 	set_input_port_default_value(2, Vector3(0.0, 0.0, 0.0));
 }
+
+////////////// Flipbook
+
+String VisualShaderNodeFlipbook::get_caption() const {
+	return "Flipbook";
+}
+
+int VisualShaderNodeFlipbook::get_input_port_count() const {
+	return 12;
+}
+
+VisualShaderNodeFlipbook::PortType VisualShaderNodeFlipbook::get_input_port_type(int p_port) const {
+	if (p_port == 0 || p_port == 1) {
+		return PORT_TYPE_SCALAR;
+	}
+	return PORT_TYPE_VECTOR;
+}
+
+String VisualShaderNodeFlipbook::get_input_port_name(int p_port) const {
+	switch (p_port) {
+		case 0:
+			return "time";
+		case 1:
+			return "texture value";
```

Comment:
I think this should be called "Frames" or "Frame Count" and accept integer numbers(I know, currently it's not possible because it's not implemented). So the number of texture inputs should depend on it.

Reply:
name "texture value" will replace "frames" or leave "texture value"?

Reply:
"texture values" is too long and no other nodes use two words inside the parameter name. This should be "frames". "frames" should be like 2.0, 3.0, 4.0 - which represents the number of frames which uses by this "book".

Reply:
The number of frames should change the amount of the input parameters, by default leave it minimal.

Reply:
I understand the visual shader is not implemented integer :)

Reply:
Well, you can use a scalar, but it's better to be an integer - so either wait to the time where this will be implemented, or implement it yourself, or leave it a scalar and hope it's good enough for approval